src/Backoffice/Flight/App/Requests/UpsertFlightRequest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Flight\App\Requests;
+
+use Carbon\CarbonImmutable;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+use Illuminate\Validation\Validator;
+use Lightit\Backoffice\Airline\Domain\Models\Airline;
+use Lightit\Backoffice\City\Domain\Models\City;
+use Lightit\Backoffice\Flight\Domain\DataTransferObject\FlightDto;
+
+class UpsertFlightRequest extends FormRequest
+{
+    public const AIRLINE_ID = 'airline_id';
+
+    public const DEPARTURE_CITY_ID = 'departure_city_id';
+
+    public const ARRIVAL_CITY_ID = 'arrival_city_id';
+
+    public const DEPARTURE_TIME = 'departure_time';
+
+    public const ARRIVAL_TIME = 'arrival_time';
+
+    /**
+     * @return array<string, mixed>
+     */
+    public function rules(): array
+    {
+        return [
+            self::AIRLINE_ID => ['required', Rule::exists(Airline::class, 'id')],
+            self::DEPARTURE_CITY_ID => ['required', Rule::exists(City::class, 'id')],
+            self::ARRIVAL_CITY_ID => ['required', Rule::exists(City::class, 'id')],
+            self::DEPARTURE_TIME => ['required', 'date_format:Y-m-d H:i:s'],
+            self::ARRIVAL_TIME => ['required', 'date_format:Y-m-d H:i:s'],
+        ];
+    }
+
+    public function after(): array
+    {
+        return [
+            function (Validator $validator): void {
+                if ($this->validateSameOriginAndDestinationCity()) {
```

Comment:
I feel like this function name is almost perfect, change it to `isOriginCitySameAsDestinationCity`, that way you know it'll return a bool since it's an `is` function